storage/src/main/java/org/apache/kafka/storage/internals/log/TimeIndex.java

```diff
@@ -75,13 +75,14 @@ public void sanityCheck() {
         TimestampOffset entry = lastEntry();
         long lastTimestamp = entry.timestamp;
         long lastOffset = entry.offset;
-        if (entries() != 0 && lastTimestamp < timestamp(mmap(), 0))
-            throw new CorruptIndexException("Corrupt time index found, time index file (" + file().getAbsolutePath() + ") has "
-                + "non-zero size but the last timestamp is " + lastTimestamp + " which is less than the first timestamp "
-                + timestamp(mmap(), 0));
+
         if (entries() != 0 && lastOffset < baseOffset())
             throw new CorruptIndexException("Corrupt time index found, time index file (" + file().getAbsolutePath() + ") has "
                 + "non-zero size but the last offset is " + lastOffset + " which is less than the first offset " + baseOffset());
+        if (entries() != 0 && lastTimestamp < timestamp(mmap(), 0))
```

Comment:
I am assuming that the reason of moving this down is to use the less expensive validation first?

Reply:
In the test we are not mocking mmap(), it is a bit trickier to do so, so when I write test to check index corruption I do it by satisfying this
```
if (entries() != 0 && lastOffset < baseOffset())
```